### PR TITLE
Intro to React Post

### PR DESCRIPTION
**Pull request by HafizMBilal009**
Added a new blog post for starting out with React

### CODE REVIEW

File: _posts/2020-02-09-intro-to-react.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+author: Hafiz Muhammad Bilal
+authorUrl: https://github.com/HafizMBilal009
+date: "2020-02-09 12:40:31"
+---
+
+### React is a very popular **JavaScript** library for building user interfaces
+
+## Pre-requisites
+
+Basic knowledge of :
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:13]:
```suggestion
Basic knowledge of :
```

Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:13]: Remove the line.

File: _posts/2020-02-09-intro-to-react.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+- HTML
+- Javascript
+
+## Why React?
+
+React was introduced by Facebook in 2013. Due to its readable syntax & code reusability, It has become one of the most popular & most used libraries for front-end development.
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:20]:
```suggestion
React was introduced by Facebook in 2013. Due to its readable syntax & code re-usability, it has become one of the most popular & most used libraries for front-end development.
```

File: _posts/2020-02-09-intro-to-react.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+React has a very strong community on the internet world so most issues during development are solved quickly.
+React is easy to learn as its JSX (Javascript XML) syntax is very close to HTML and enhances the confidence of a novice developer.
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:22]: The bullets should not be ended with a full-stop.

File: _posts/2020-02-09-intro-to-react.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+## Basic Concepts of React
+
+We will discuss some basic fundamental concepts of React :
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:26]:
```suggestion
We will discuss some basic fundamental concepts of React:
```

File: _posts/2020-02-09-intro-to-react.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+### 1. Components
+
+Components are building blocks of a React app. A component has one or more than one element of HTML to re-use it wherever needed. With the use of components, we get rid of writing redundant code. There are two types of components i.e Class Components and Functional Components
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:30]: Class Components and Functional Components can be put in bullets here.

File: _posts/2020-02-09-intro-to-react.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+### 2. JSX
+
+JSX is a way to write React code. As from it's name we can understand it's a mixture of Javascript and XML as a result of a syntax which is easy to understand.
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:34]:
```suggestion
JSX is a way to write React code. As from its name we can understand it's a mixture of JavaScript and XML as a result of a syntax which is easy to understand.
```

File: _posts/2020-02-09-intro-to-react.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+For example :
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:35]:
```suggestion
For example:
```

File: _posts/2020-02-09-intro-to-react.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+```jsx
+const element = <h1>Welcome to React World!</h1>;
+```
+
+### 3. Props
+
+Props are arguments we can pass into React components. Basically props are used for connecting different components by passing some as an attribute to a React component.
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:43]:
```suggestion
Props are arguments we can pass into React components. Basically props are used for connecting different components by passing some value as an attribute to a React component.
```

File: _posts/2020-02-09-intro-to-react.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+One component passes a value to the other one and other component can use this value in it.
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:44]:
```suggestion
One component can pass value in the other.
```

File: _posts/2020-02-09-intro-to-react.md [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+
+### 4. State
+
+State is the heart of a React app. Basically, state is used to store data which is required in our React component. Using state, changes in data can be seen in real-time without refreshing webpage.
+
+## Enough Talk, Let's Code
+
+We will build a simple React app which shows a list of quotes with their authors. We can add a quote and see a live update in the quotes table.
+Let's start by creating a simple React app by running this command :
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:53]:
```suggestion
Let's start by creating a simple React app by running this command:
```

File: _posts/2020-02-09-intro-to-react.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+}
+.add-quote-form input {
+  width: 100%;
+  margin-top: 0.25rem;
+  margin-bottom: 0.25rem;
+  height: 1.5rem;
+}
+```
+
+You can get this code on [Github](https://github.com/HafizMBilal009/my-app.git)
```
Comment (fahadsiddiqui) [inline _posts/2020-02-09-intro-to-react.md:213]:
```suggestion
[Here](https://github.com/HafizMBilal009/my-app.git) is the code repository you can use following this blog.
```

**Review by fahadsiddiqui — CHANGES_REQUESTED** [review]

**Review by mzaeemz — APPROVED** [review]
LGTM

**Review by fahadsiddiqui — APPROVED** [review]

**Pull request merged**